File: modules/coupon-domain/src/main/java/com/thanh0x/coursedeal/model/coupon/CourseLevel.java
package com.thanh0x.coursedeal.model.coupon;

import lombok.Getter;

/**
 * Enum representing the difficulty level of a course.
 */
@Getter
public enum CourseLevel {
    BEGINNER("Beginner"),
    INTERMEDIATE("Intermediate"),
    EXPERT("Expert"),
    ALL_LEVELS("All Levels");

    private final String displayName;

    CourseLevel(String displayName) {
        this.displayName = displayName;
    }

    public static CourseLevel fromString(String text) {
        for (CourseLevel level : CourseLevel.values()) {
            if (level.displayName.equalsIgnoreCase(text) || level.name().equalsIgnoreCase(text)) {
                return level;
            }
        }
        return ALL_LEVELS; // Default
    }
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/model/coupon/CourseLevel.kt
package com.thanh0x.coursedeal.model.coupon

/**
 * Enum representing the difficulty level of a course.
 */
enum class CourseLevel(val displayName: String) {
    BEGINNER("Beginner"),
    INTERMEDIATE("Intermediate"),
    EXPERT("Expert"),
    ALL_LEVELS("All Levels"),
    ;

    companion object {
        fun fromString(text: String?): CourseLevel {
            if (text == null) return ALL_LEVELS
            return entries.find {
                it.displayName.equals(text, ignoreCase = true) || it.name.equals(text, ignoreCase = true)
            } ?: ALL_LEVELS
        }
    }
}
